draft-ietf-tokbind-https-06.xml

```diff
@@ -720,6 +720,29 @@ contexts. Other approaches are possible, but are outside the scope of this speci
       
     </section>
 
+    <section title="Implementation Considerations" anchor="impl-cons">
+      <t>
+	In addition to the functionality to use provided and referred Token Bindings
+	from Web User Agents defined by this specification,
+	the ability to supply provided, referred, and possibly other Token Binding values
+	is also intended to be made available to applications not using User Agents,
+	such as native applications.
+	Thus, Token Binding implementations providing an Application Programming Interface (API) should 
+	provide means for applications to produce Token Binding messages containing Token Binding values
+	for application-specified TLS connections,
+	such as by identifying the connections or by supplying the corresponding EKM values.
+      </t>
+      <t>
+	The application needs to be able to denote the Token Binding type (e.g., provided, referred, etc.)
+	to be used in the message for each specified connection (or EKM value),
+	as well as the TLS connection the application will send the Token Binding message over.
+	The result should be a Token Binding message containing Token Binding values
+	bound to an application-specified TLS connection over which the Token Binding message is sent.
+	Additionally, see <xref target="privacy-cons"/> <xref target="privacy-cons" format="title"/>,
+	which provides further guidance regarding the use of this functionality.
+      </t>
+    </section>
+
```

Comment:
Thanks for making these mods, Mike!

regarding the above section..

hmm, this spec is overall written in terms of "client", rather than "(web) user agent". E.g., the latter term is first introduced in the section preceding this one and occurs only a few times. So I think we can simplify the first paragraph accordingly. Plus, the second paragraph essentially repeats info that's in the first. 

how about the below build on the above?

```
<t>
  HTTPS-based applications may have multi-party use 
  cases other than, or in addition to, the HTTP redirect-based 
  signaling-and-conveyance of referred token bindings, as 
  presented above in <xref target="sctn-http-redir">.
</t>
<t>
  Thus, generic Token Binding implementations intended to support 
  any HTTPS-based client-side application (e.g., so-called "native 
  applications"), should provide means for applications to have 
  Token Binding messages, containing Token Binding IDs of various 
  application-specified Token Binding types and 
  for application-specified TLS connections, conveyed over an
  application-specified HTTPS connection, i.e., within the 
  TokenBindingMessage conveyed by the Sec-Token-Binding header
  field. 
  <list style="hanging" hangIndent="7">
    <t hangText="NOTE:">
      See <xref target="privacy-cons"/> 
      "<xref target="privacy-cons" format="title"/>",
      for privacy guidance regarding the use of 
      this functionality.
     </t>
  </list>
</t>
```

[ fyi/fwiw, the markup for the NOTE is as I used in RFC6797 and passed muster with the rfc-editor ]